File: server/src/main/java/com/content/security/entity/Vaccineordervaccine.java
package com.content.security.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "vaccineordervaccine")
public class Vaccineordervaccine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "quentity")
    private Integer quentity;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "vaccineorder_id", nullable = false)
    private Vaccineorder vaccineorder;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "vaccine_id", nullable = false)
    private Vaccine vaccine;

}

File: client/src/app/core/entity/vaccineordervaccine.ts
import {Product} from "./product";
import {Vaccine} from "./vaccine";

export class VaccineOrderVaccine{
  id:number
  vaccine:Vaccine
  quentity:number


  constructor(id:number,vaccine:Vaccine,quentity:number) {
    this.id = id;
    this.vaccine = vaccine;
    this.quentity = quentity;


  }
}
